collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/client/JiraXRayRestClientSupplier.java

```
@@ -0,0 +1,163 @@
+	 * username/password map
+	 *
+	 * @param jiraBasicAuthCredentialsInBase64
+	 *            Base64-encoded single string in the following format:
+	 *            <em>username:password</em><br/>
+	 * <br/>
+	 *            A null parameter value will result in an empty hash map
+	 *            response (e.g., nothing gets decoded)
+	 * @return Decoded username/password map of strings
+	 */
+	private Map<String, String> decodeCredentials(String jiraBasicAuthCredentialsInBase64) {
+		Map<String, String> credMap = new LinkedHashMap<String, String>();
+		if (jiraBasicAuthCredentialsInBase64 != null) {
+			//the tokenize includes a \n to ensure we trim those off the end (mac base64 adds these!)
+			StringTokenizer tokenizer = new StringTokenizer(new String(
+					Base64.decodeBase64(jiraBasicAuthCredentialsInBase64)), ":\n");
+			for (int i = 0; tokenizer.hasMoreTokens(); i++) {
+				if (i == 0) {
+					credMap.put("username", tokenizer.nextToken());
+				} else {
+					credMap.put("password", tokenizer.nextToken());
+				}
+			}
+		}
+
+		return credMap;
+
+	}
+
+	/**
+	 * Generates an authenticated proxy connection URI and Jira URI for use in
+	 * talking to Jira.
+	 *
+	 * @param jiraBaseUri
+	 *            A string representation of a Jira URI
+	 * @param fullProxyUrl
+	 *            A string representation of a completed proxy URL:
+	 *            http://your.proxy.com:8080
+	 * @param username
+	 *            A string representation of a username to be authenticated
+	 * @param password
+	 *            A string representation of a password to be used in
+	 *            authentication
+	 * @return A fully configured Jira URI with authenticated proxy connection
+	 */
+	private URI createJiraConnection(String jiraBaseUri, String fullProxyUrl, String username,
+									 String password) {
+		final String uname = username;
+		final String pword = password;
+		Proxy proxy = null;
+		URLConnection connection = null;
+		try {
+			if (!StringUtils.isEmpty(jiraBaseUri)) {

```

The nested ifs here are too many. Consider the concept for fail fast